src/extension-runners/firefox-android.js

```diff
@@ -122,15 +123,29 @@ export class FirefoxAndroidExtensionRunner {
       adbHost,
       adbPort,
       ADBUtils = DefaultADBUtils,
+      cleanArtifacts,
     } = this.params;
 
+
     this.adbUtils = new ADBUtils({
       adbBin, adbHost, adbPort,
     });
+    // Call all the registered cleanup callbacks.
+    for (const fn of this.cleanupCallbacks) {
+      try {
+        fn();
+      } catch (error) {
+        log.error(error);
+      }
+    }
```

Comment:
what is the purpose of calling the cleanup callbacks when web-ext is just starting the firefox-android extension runner?
it looks wrong, this isn't going to cleanup old artifacts dir on the target device and it is not going to do anything useful for that purpose.